lib/ramble/ramble/cmd/results.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022-2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import llnl.util.tty as tty
+import json
+
+import ramble.experimental.uploader
+
+
+description = "take actions on experiment results"
+section = "results"
+level = "short"
+
+subcommands = [
+    'upload',
+]
+
+#: Dictionary mapping subcommand names and aliases to functions
+subcommand_functions = {}
+
+
+def results_upload_setup_parser(subparser):
+    subparser.add_argument(
+        'filename', metavar='filename',
+        help='path of file to upload')
+
+
+def results_upload(args):
+    """Imports Ramble experiment results from JSON file and uploads them."""
+    imported_results = import_results_file(args.filename)
+
+    ramble.experimental.uploader.upload_results(imported_results)
+
+
+def import_results_file(filename):
+    """
+    Import Ramble experiment results from a JSON file.
+    """
+    tty.debug("File to import:")
+    tty.debug(filename)
+
+    imported_file = open(filename)
+
+    try:
+        tty.msg("Import file...")
+        parsed_json_file = json.load(imported_file)
+        # Check if data contains an experiment
+        if parsed_json_file.get('experiments'):
+            return parsed_json_file
+        else:
+            tty.die("Error parsing file: Does not contain valid data to upload.")
+    except ValueError:
+        tty.die("Error parsing file: Invalid JSON formatting.")
+
+
+def setup_parser(subparser):
+    sp = subparser.add_subparsers(metavar='SUBCOMMAND',
+                                  dest='results_command')
+
+    for name in subcommands:
+        if isinstance(name, (list, tuple)):
+            name, aliases = name[0], name[1:]
```

Comment:
I know you copied this from elsewhere, but can we remove this unneeded alias stuff? 

We don't need aliases here (yet?), but when we do it feels like we would want to pull this out into a common function somewhere than repeat it for a third time

Reply:
Reworked this to use the parser setup pattern from repo.py